apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/shard/helper/CsvExporterTest.java

```diff
@@ -0,0 +1,274 @@
+        doReturn(UUID.fromString("a2e9b946-290b-48b6-9985-dc2e5a5860a1")).when(chain).getChainId();
+        // init several derived tables
+        AccountCurrencyTable.getInstance().init();
+        Account.init(extension.getDatabaseManger(), propertiesHolder, null, null, blockchain, null, null, accountTable);
+        AccountInfoTable.getInstance().init();
+        Alias.init();
+//        PhasingOnly.get(Long.parseUnsignedLong("7995581942006468815")); // works OK!
+//        PhasingOnly.get(Long.parseUnsignedLong("2728325718715804811")); // error, doesn't load from db !!
+//        PhasingOnly.get(Long.parseLong("-8446384352342482748")); // error, doesn't load from db !!
+        PhasingOnly.get(Long.parseLong("-4013722529644937202")); // works OK!
+        AccountAssetTable.getInstance().init();
+        PublicKeyTable publicKeyTable = new PublicKeyTable(blockchain);
+        publicKeyTable.init();
+        AccountLedgerTable accountLedgerTable = new AccountLedgerTable();
+        accountLedgerTable.init();
+        AccountGuaranteedBalanceTable accountGuaranteedBalanceTable = new AccountGuaranteedBalanceTable(blockchainConfig, propertiesHolder);
+        accountGuaranteedBalanceTable.init();
+        DGSPurchaseTable purchaseTable = new DGSPurchaseTable();
+        purchaseTable.init();
+    }
+
+    @Test
+    void exportDerivedTables() {
+        doReturn(temporaryFolderExtension.newFolder("csvExport").toPath()).when(dirProvider).getDataExportDir();
+//        doReturn(createPath("csv-export")).when(dirProvider).getDataExportDir(); // prod data test
+        cvsExporter = new CsvExporterImpl(dirProvider.getDataExportDir(), extension.getDatabaseManger(), shardDaoJdbc);
+        assertNotNull(cvsExporter);
+
+        Collection<DerivedTableInterface> result = registry.getDerivedTables(); // extract all derived tables
+        int targetHeight = 8000;
+//        int targetHeight = 2_000_000; // prod data test
+        int batchLimit = 1; // used for pagination and partial commit
+        int[] tablesWithDataCount = new int[1]; // some table doesn't have exported data
+
+        long start = System.currentTimeMillis();
+        result.forEach(item -> {
+            long start2 = System.currentTimeMillis();
+            long exportedRows = 0;
+            exportedRows = cvsExporter.exportDerivedTable(item, targetHeight, batchLimit);
+            if (exportedRows > 0) {
+                tablesWithDataCount[0] = tablesWithDataCount[0] + 1;
+            }
+            log.debug("Processed Table = {}, exported = '{}' rows in {} secs", item, exportedRows, (System.currentTimeMillis() - start2) / 1000 );
+        });
+        log.debug("Processed Tables = [{}] in {} sec", result.size(), (System.currentTimeMillis() - start) / 1000 );
+        String[] extensions =  new String[]{"csv"};
+        Collection filesInFolder = FileUtils.listFiles(dirProvider.getDataExportDir().toFile(), extensions, false) ;
+        assertNotNull(filesInFolder);
```

Comment:
Please, make more assertions, such test verify that files were created, but dont check consistency of exported values